lib/metadataTypes/FilterDefinition.js

```diff
@@ -0,0 +1,279 @@
+                metadata.categoryId,
+                'ID',
+                'Path'
+            );
+            delete metadata.categoryId;
+
+            if (metadata.derivedFromType === 2) {
+                // DataExtension
+                metadata.r__dataExtension_CustomerKey = Util.getFromCache(
+                    this.cache,
+                    'dataExtension',
+                    metadata.derivedFromObjectId,
+                    'ObjectID',
+                    'CustomerKey'
+                );
+            }
+            delete metadata.derivedFromObjectId;
+            delete metadata.derivedFromType;
+            metadata.c__filterDefinition = await xml2js.parseStringPromise(
+                metadata.filterDefinitionXml /* , options */
+            );
+
+            // TODO check if Condition ID needs to be resolved or can be ignored
+        } catch (ex) {
+            Util.logger.error(
+                `FilterDefinition '${metadata.name}' (${metadata.key}): ${ex.message}`
+            );
+        }
+        return metadata;
+    }
+    /**
+     * prepares a item for deployment
+     * @param {FilterDefinitionItem} metadata a single record
+     * @returns {Promise<FilterDefinitionItem>} Promise of updated single item
+     */
+    static async preDeployTasks(metadata) {
+        // folder
+        metadata.categoryId = Util.getFromCache(
+            this.cache,
+            'folder',
+            metadata.r__folder_Path,
+            'Path',
+            'ID'
+        );
+        delete metadata.r__folder_Path;
+
+        if (metadata.derivedFromObjectTypeName === 'SubscriberAttributes') {
+            // List
+            metadata.derivedFromType = 1;
+            metadata.derivedFromObjectId = '00000000-0000-0000-0000-000000000000';
+        } else {
+            // DataExtension
+            metadata.derivedFromType = 2;
+
+            if (metadata.r__dataExtension_CustomerKey) {
+                metadata.derivedFromObjectId = Util.getFromCache(
+                    this.cache,
+                    'dataExtension',
+                    metadata.r__dataExtension_CustomerKey,
+                    'CustomerKey',
+                    'ObjectID'
+                );
+                delete metadata.r__dataExtension_CustomerKey;
+            }
+        }
+        delete metadata.c__filterDefinition;
+        delete metadata.c__soap_DataFilter;
+
+        return metadata;
+    }
+    /**
+     * Creates a single item
+     * @param {FilterDefinitionItem} metadata a single item
+     * @returns {Promise<FilterDefinitionItem>} Promise
+     */
+    static create(metadata) {
+        // TODO test the create
+        return super.createREST(metadata, '/email/v1/filters/filterdefinition/');
+    }
+    /**
+     * Updates a single item
+     * @param {FilterDefinitionItem} metadata a single item
+     * @returns {Promise<FilterDefinitionItem>} Promise
+     */
+    static update(metadata) {
+        // TODO test the update
```

Comment:
if its not tested, then wouldnt it be better to not have this in the PR?

Reply:
in my defense - the PR is in draft mode still.
![image](https://user-images.githubusercontent.com/1917227/149722276-3b4e1189-3408-445d-bc38-5910657474aa.png)